src/logging.cpp

```diff
@@ -225,8 +218,14 @@ std::string BCLog::Logger::LogLevelToStr(BCLog::Level level)
     assert(false);
 }
 
-std::string LogCategoryToStr(BCLog::LogFlags category)
+static std::string LogCategoryToStr(BCLog::LogFlags category)
 {
+    if (category == BCLog::NONE) {
+        return "";
+    }
```

Comment:
I am fine either way. The only caller of `LogCategoryToStr()` has checked explicitly that it is not `NONE`. This was suggested here: https://github.com/bitcoin/bitcoin/pull/29798#discussion_r1553715178, @ryanofsky, what do you think?

Reply:
I would still prefer the assertion, but if we're going to return a string, "none" is probably better than an empty string. But it's a minor point either way.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/29798#discussion_r1693713026

Good catch. Since this case can't happen and the function is already written to assert on other invalid categories, agree it's better to remove this unused special case. I think it might be overkill to add an assert for this since there is an assert right below that will catch it, but no real opinion on that.

Reply:
Alright, even better! I just removed that `if`, no need for two asserts.